src/Nethermind/Nethermind.Evm/CodeAnalysis/CMSketch.cs

```diff
@@ -0,0 +1,83 @@
+
+using System;
+using System.Collections.Concurrent;
+
+namespace Nethermind.Evm.CodeAnalysis
+{
+
+    public class CMSketch
+    {
+        private ConcurrentDictionary<(int Row, int Column), uint> _sketch;
+        private int[] _seeds;
+        private int _buckets;
+        private int _numberOfhashFunctions;
+
+        public CMSketch(int numberOfhashFunctions, int buckets)
+        {
+            _sketch = new ConcurrentDictionary<(int, int), uint>();
+            _buckets = buckets;
+            _numberOfhashFunctions = numberOfhashFunctions;
+            _seeds = new int[numberOfhashFunctions];
+            Random rand = new Random();
+            for (int i = 0; i < numberOfhashFunctions; i++)
+            {
+                _seeds[i] = rand.Next(int.MinValue, int.MaxValue);
+            }
+        }
+
+        public void Update(ulong item)
+        {
+            for (int hasher = 0; hasher < _numberOfhashFunctions; hasher++)
+            {
+                _sketch.AddOrUpdate((hasher, ComputeHash(item, hasher, _seeds, _numberOfhashFunctions) % _buckets),
+                             1,
+                         (key, value) => value + 1
+                        );
+            }
+        }
+
+        public uint Query(ulong item)
+        {
+            var minCount = uint.MaxValue;
+            for (int hasher = 0; hasher < _numberOfhashFunctions; hasher++)
+            {
+                _sketch.TryGetValue((hasher, ComputeHash(item, hasher, _seeds, _numberOfhashFunctions) % _buckets), out uint count);
+                minCount = Math.Min(minCount, count);
+            }
+            return minCount;
+        }
+
+
+        public uint UpdateAndQuery(ulong item)
+        {
+            Update(item);
+            return Query(item);
+        }
+
+        public void Reset()
+        {
+            _sketch = new ConcurrentDictionary<(int, int), uint>();
+        }
+
+        public static int ComputeHash(ulong value, int hasher, int[] _seeds, int _breadth)
```

Comment:
Why is this static? Is it required to be? Would be nice to have the `_seeds` and `_breadth` got from `this`? Also, assuming that the `hashes` is within a range (the call in `Update` and `Query`) do we need to do the modulo in the implementation?

Reply:
TODO